server/e2e/jobs/specs/library.e2e-spec.ts

```diff
@@ -236,6 +165,185 @@ describe(`${LibraryController.name} (e2e)`, () => {
+        expect.arrayContaining([
+          expect.objectContaining({
+            isOffline: true,
+            originalFileName: 'silver_fir.jpg',
+          }),
+          expect.objectContaining({
+            isOffline: true,
+            originalFileName: 'tanners_ridge.jpg',
+          }),
+        ]),
+      );
+    });
+
+    it('should mark a rediscovered file as back online', async () => {
+      await fs.promises.cp(`${IMMICH_TEST_ASSET_PATH}/albums/nature`, `${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature`, {
+        recursive: true,
+      });
+
+      const library = await api.libraryApi.create(server, admin.accessToken, {
+        ownerId: admin.userId,
+        type: LibraryType.EXTERNAL,
+        importPaths: [`${IMMICH_TEST_ASSET_TEMP_PATH}`],
+      });
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const onlineAssets = await api.assetApi.getAllAssets(server, admin.accessToken);
+      expect(onlineAssets.length).toBeGreaterThan(1);
+
+      await fs.promises.rm(`${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature/silver_fir.jpg`);
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      await fs.promises.cp(`${IMMICH_TEST_ASSET_PATH}/albums/nature`, `${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature`, {
+        recursive: true,
+      });
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const assets = await api.assetApi.getAllAssets(server, admin.accessToken);
+
+      expect(assets).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            isOffline: false,
+            originalFileName: 'silver_fir.jpg',
+          }),
+          expect.objectContaining({
+            isOffline: false,
+            originalFileName: 'tanners_ridge.jpg',
+          }),
+        ]),
+      );
+    });
+
+    it('should mark a rediscovered file in import paths as back online', async () => {
+      await fs.promises.cp(`${IMMICH_TEST_ASSET_PATH}/albums/nature`, `${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature`, {
+        recursive: true,
+      });
+
+      await fs.promises.mkdir(`${IMMICH_TEST_ASSET_TEMP_PATH}/albums/other`);
+
+      const library = await api.libraryApi.create(server, admin.accessToken, {
+        ownerId: admin.userId,
+        type: LibraryType.EXTERNAL,
+        importPaths: [`${IMMICH_TEST_ASSET_TEMP_PATH}/albums/`],
+      });
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const initialAssets = await api.assetApi.getAllAssets(server, admin.accessToken);
+      expect(initialAssets.length).toBeGreaterThan(1);
+
+      await api.libraryApi.setImportPaths(server, admin.accessToken, library.id, [
+        `${IMMICH_TEST_ASSET_TEMP_PATH}/albums/other`,
+      ]);
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const offlineAssets = await api.assetApi.getAllAssets(server, admin.accessToken);
```

Comment:
Same here and a few lines below. I'm aware that it's currently hard to test these things as there aren't any websocket events for job/queue completion atm. But still, you shouldn't just write these tests as if they'd happen linearly because that (1) doesn't really test anything and (b) can result in flaky tests. In my opinion these tests don't add any value currently since they're simply not reliable.